### PR TITLE
Adding basic telemetry instrumentation for glimpse client

### PR DESCRIPTION
**Pull request by mike-kaufman**
- added a TelemetyClient class that will hook up to various shell events and send corresponding App Insights telemetry events. 
 - Pull the telemetry configuration data from a "telemetry-config" resource. 
 - Added "fake data" to support dev-time scenarios.

### CODE REVIEW

File: src/lib/TelemetryClient.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    constructor() {
+        this.sessionId = uuid.v4();
+        const self = this;
+        metadataRepository.registerListener((metadata) => {
+
+            if (!metadata.resources || !metadata.resources['telemetry-config']) {
+                self.isTelemetryEnabled = false;
+            }
+            else {
+                const uri = metadata.resources['telemetry-config'];
```
Comment (avanderhoorn) [inline src/lib/TelemetryClient.ts]: Now that i think about it, I wonder if to save having to make a second request for this data, we could include this data in the metadata payload...

Comment (mike-kaufman) [inline src/lib/TelemetryClient.ts]: It's only one extra request, so it isn't like it's repeated.  That said, I don't have strong opinions about it, other than what I have is implemented & seems to be working.

**Comment by mike-kaufman** [comment]
ping... I"d like to get this in.

File: src/lib/TelemetryClient.ts (hunk 1 above)
Comment (avanderhoorn) [inline src/lib/TelemetryClient.ts]: Thats fine for now

**Pull request merged**